### DISCUSSION TITLE
Erro tanto lara 8 quanto lara 5.4

### DISCUSSION THREAD

**Discussion by ThiagoSilvaCleanHouse** [jetimob/juno-sdk-php-laravel#24] [Q&A]
Olá, 

Qual e versão do Lara a ser utilizada?

Em ambos deu o seguinte erro:

[2021-11-04 16:00:47] local.ERROR: Symfony\Component\Debug\Exception\FatalThrowableError: Parse error: syntax error, unexpected 'string' (T_STRING), expecting function (T_FUNCTION) or const (T_CONST) in C:\Users\thiag\Documents\Clean House Express\Sistema\new_project_app\clean_v1\vendor\jetimob\juno-sdk-php-laravel\src\Lib\Model\Billing.php:12
Stack trace:

Podeira por gentileza, me ajudar?

Att,

**Comment by alanwgt** [comment]
Olá, Thiago!

Cara, utilizamos o pacote na versão 8 do laravel sem problemas por aqui.

Qual a versão do PHP que tu está executando?

> Lembrando que a versão precisa ser >= 7.4

**Reply by ThiagoSilvaCleanHouse** [reply]
Estava com com:  "php": "^7.3|^8.0",

Mas mudei para: 
"php": "^7.3", -> Fiz um composer install, mas ainda assim ao tentar instalar, da o seguinte erro: 

PS C:\Users\thiag\source\repos\clin_api_lara8> composer require jetimob/juno-sdk-php-laravel


  [InvalidArgumentException]
  Package jetimob/juno-sdk-php-laravel has a PHP requirement incompatible with your PHP version, PHP extensions and Composer version  
                                                                                                                                      

require [--dev] [--dry-run] [--prefer-source] [--prefer-dist] [--fixed] [--no-suggest] [--no-progress] [--no-update] [--no-install] [--no-scripts] [--update-no-dev] [-w|--update-with-dependencies] [-W|--update-with-all-dependencies] [--with-dependencies] [--with-all-dependencies] [--ignore-platform-req IGNORE-PLATFORM-REQ] [--ignore-platform-reqs] [--prefer-stable] [--prefer-lowest] [--sort-packages] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--apcu-autoloader-prefix APCU-AUTOLOADER-PREFIX] [--] [<packages>]...

PS C:\Users\thiag\source\repos\clin_api_lara8> php -v
PHP 7.3.28 (cli) (built: Apr 27 2021 17:19:33) ( ZTS MSVC15 (Visual C++ 2017) x64 )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.28, Copyright (c) 1998-2018 Zend Technologies

**Reply by ThiagoSilvaCleanHouse** [reply]
Forcei a Instalação com:
composer require --ignore-platform-reqs  jetimob/juno-sdk-php-laravel

ai a instalação vai ok. 

Porém da erro na models do proejto, começando no Billing: 

Parse error: syntax error, unexpected 'string'


Na arquivo:
vendor\jetimob\juno-sdk-php-laravel\src\Lib\Model\Billing.php:12

**Comment by alanwgt** [comment]
Tu está utilizando uma versão não compatível com a versão mínima (`PHP 7.3.28 (cli) (built: Apr 27 2021 17:19:33) (ZTS MSVC15 (Visual C++ 2017) x64`).

As tipagens de variáveis dentro de classes (propriedades) só foram implementadas no [PHP 7.4](https://wiki.php.net/rfc/typed_properties_v2).

Teus problemas serão resolvidos atualizando a versão do PHP para a 7.4  :)

**Reply by ThiagoSilvaCleanHouse** [reply]
Eu que tinha baixado pois tinha entendido errado, achei que deveria ser abaixo do 7.4
Mas, tive de mudar o Xammp para a versão com php 7.4
Agora de certo. 

Obrigado.

**Reply by alanwgt** [reply]
Sem problemas!

Precisando de ajuda, fique a vontade para abrir uma nova discussão.
